pkg/controller/pubsub/pubsub_init.go

```diff
@@ -89,31 +90,45 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 }
 
 func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
-	providers := psClient.Providers()
-	if len(providers) == 0 {
-		log.Info("No pub sub tool is enabled")
-		return reconcile.Result{}, nil
+	log.Info("Reconcile", "request", request, "namespace", request.Namespace, "name", request.Name)
+
+	deleted := false
+	config := &corev1.ConfigMap{}
+	err := r.Get(ctx, request.NamespacedName, config)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return reconcile.Result{}, err
+		}
+		deleted = true
 	}
 
-	if newConnection, ok := providers[request.Name]; ok {
-		config := &corev1.ConfigMap{}
-		err := r.Get(ctx, request.NamespacedName, config)
+	if deleted {
+		err := r.system.CloseConnection(request.Name)
 		if err != nil {
 			return reconcile.Result{}, err
 		}
-
-		if len(config.Data) == 0 {
-			return reconcile.Result{}, fmt.Errorf(fmt.Sprintf("data missing in configmap %s, unable to configure respective pubsub", request.NamespacedName))
+		log.Info("removed connection", "name", request.Name)
+	} else {
+		providers := psClient.Providers()
+		if len(providers) == 0 {
+			log.Info("No pub sub tool is enabled")
+			return reconcile.Result{}, nil
 		}
-		log.Info(fmt.Sprintf("system state before initiating connection %+v", r.system))
-		err = r.system.InitiateConnection(ctx, newConnection, config.Data)
+
 		if err != nil {
 			return reconcile.Result{}, err
 		}
-		log.Info(fmt.Sprintf("Connection established %+v", r.system))
+		if len(config.Data) == 0 {
+			return reconcile.Result{}, fmt.Errorf(fmt.Sprintf("data missing in configmap %s, unable to configure respective pubsub", request.NamespacedName))
+		}
 
-		log.Info("Pub sub clients are initialized without error")
+		if newConnection, ok := providers[config.Data["provider"]]; ok {
```

Comment:
What happens if the connection is unrecognized? Is this an error that gets reported to the user? Should it be?

Reply:
As we discussed, we will log it for now and then if/when we implement a CRD for this, we can include it in status